includes/class-newspack-popups.php

```diff
@@ -573,5 +574,95 @@ public static function api_config_missing_notice() {
 			</div>
 		<?php
 	}
+
+	/**
+	 * Delete campaign.
+	 *
+	 * @param int $id Campaign ID.
+	 */
+	public static function delete_campaign( $id ) {
+		wp_delete_term( $id, self::NEWSPACK_POPUPS_TAXONOMY );
+	}
+
+	/**
+	 * Duplicate campaign.
+	 *
+	 * @param int    $id Campaign ID.
+	 * @param string $name New campaign name.
+	 */
+	public static function duplicate_campaign( $id, $name ) {
+		$term = wp_insert_term( $name, self::NEWSPACK_POPUPS_TAXONOMY );
+		if ( is_wp_error( $term ) ) {
+			$term = get_term_by( 'name', $name, self::NEWSPACK_POPUPS_TAXONOMY );
+		}
+		$term = (object) $term;
+		$args = [
+			'post_type'      => self::NEWSPACK_POPUPS_CPT,
+			'posts_per_page' => 100,
+			'fields'         => 'ids',
+			'post_status'    => [ 'publish', 'pending', 'draft', 'future' ],
+			'tax_query'      => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+				[
+					'taxonomy' => self::NEWSPACK_POPUPS_TAXONOMY,
+					'field'    => 'term_id',
+					'terms'    => [ $id ],
+				],
+			],
+		];
+
+		$query = new WP_Query( $args );
+		foreach ( $query->posts as $id ) {
+			wp_set_post_terms( $id, $term->term_id, self::NEWSPACK_POPUPS_TAXONOMY, true );
+		}
+	}
+
+	/**
+	 * Rename campaign.
+	 *
+	 * @param int    $id Campaign ID.
+	 * @param string $name New campaign name.
+	 */
+	public static function rename_campaign( $id, $name ) {
+		wp_update_term(
+			$id,
+			self::NEWSPACK_POPUPS_TAXONOMY,
+			[
+				'name' => $name,
+			]
+		);
+	}
+
+	/**
+	 * Archive campaign.
+	 *
+	 * @param int  $id Campaign ID.
+	 * @param bool $status Whether to archive or unarchive (true = archive, false = unarchive).
+	 */
+	public static function archive_campaign( $id, $status ) {
+		update_term_meta( $id, self::NEWSPACK_POPUPS_TAXONOMY_STATUS, $status ? 'archive' : '' );
+	}
+
+	/**
+	 * Retrieve campaign groups.
```

Comment:
Shouldn't that be called "Retrieve campaign"?

Reply:
You're right, fixed the comment in https://github.com/Automattic/newspack-popups/pull/427/commits/08dfdd63f551a71448272dd845c338a56459d331. I'm hesitant to start renaming functions in `newspack-popups` for fear we'll create some regressions, but once the UI redesign lands we should go back and clean everything up.